src/math/Kconfig

```diff
@@ -22,15 +22,35 @@ config POWER_FIXED
 
 
 config BINARY_LOGARITHM_FIXED
-       bool "Binary Logarithm function"
-       default n
-       help
+	bool "Binary Logarithm function"
+	default n
+	help
 	  This option builds Binary Logarithm function (log2n) is the logarithm to
 	  the base 2. The binary logarithm of x is the power to which the number 2
 	  must be raised to obtain the value x. Base 2 logarithm function is implemented
 	  with a short lookup table. (log2n) operates for a range of 32 bit width size
 	  i.e. 1 to 4294967295.
 
+config NATURAL_LOGARITHM_FIXED
+	bool "Natural Logarithm function"
+	default n
+	help
+	  This option builds Natural Logarithm function (loge(N)) is the logarithm to
+	  the base e. The natural logarithm of x is the power to which the number e
+	  must be raised to obtain the value x. Base e logarithm function is implemented
+	  with a short lookup table. (loge(N)) operates for a range of 32 bit width size
```

Comment:
implementation details should not be part of config help but rather comment in the c file

Reply:
Sure, Ideally I would also like to keep description close to 1 or 2 liner. Unfortunately, I get stuck with WARNING pasted below. So I am forced to add 3-4 liner description
`WARNING: please write a paragraph that describes the config symbol fully`
Please share suggestion's , what exactly is expected here and should be followed as a rule. Thank you

Reply:
https://www.kernel.org/doc/html/latest/dev-tools/checkpatch.html
> Checkpatch is not always right. Your judgement takes precedence over checkpatch messages. If your code looks better with the violations, then its probably best left alone.

Reply:
Fully as in what it does in this context, now how it does it. That is what the source and tests are for

Reply:
any  change expected here ?

Reply:
I think @cujomalainey is saying:
- WHAT it does = the externally visible output belongs to Kconfig, 
- HOW it is done = the internal, invisible way to do it belongs to the source code (but don't just paraphrase and duplicate the code, I digress)

Consider your "... is implemented with... " sentence: is that visible outside the function?

Reply:
Marc is correct, we should not be discussing implementation here